### PR TITLE
[ENHANCEMENT] Ensure consistent usage of the Canonical Execution Engine API through the use of the ExecutionEngine.get_compute_domain() method.

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
This work removes deviations from such key interfaces as `ExecutionEngine.get_compute_domain()` in client classes in the `metrics` focused sections, and also cleans up the `active_batch_id` related methods in the `Validator` class.  These inefficiencies were identified as a result of working on a SPIKE aimed at researching the usage of `batch_id` in `domain_kwargs`.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [ENHANCEMENT], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/metrics/table_metrics/table_column_types.py [hunk 1]
```diff
@@ -47,21 +47,12 @@ def _sqlalchemy(
...
-                    "batch_id could not be determined from domain kwargs and no active_batch_data is loaded into the "
-                    "execution engine"
-                )
-        batch_data = execution_engine.loaded_batch_data_dict.get(batch_id)
-        if batch_data is None:
-            raise GreatExpectationsError(
-                "the requested batch is not available; please load the batch into the execution engine."
-            )
-        return _get_sqlalchemy_column_metadata(execution_engine.engine, batch_data)
+        batch_data, _, _ = execution_engine.get_compute_domain(
```
Comment (anthonyburdi) [inline great_expectations/expectations/metrics/table_metrics/table_column_types.py:50]: 👍 The above error checking is already contained in `get_compute_domain()`

**Review by anthonyburdi — APPROVED** [review]
LGTM - you may wish to have someone more familiar with the sqlalchemy execution engine API weigh in.

File: great_expectations/expectations/metrics/table_metrics/table_column_types.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/metrics/table_metrics/table_column_types.py:50]: @anthonyburdi Yes -- and even more than error checking.  This is moving the metric implementation to use the correct interface -- the very subject of this PR.  Thanks!

**Comment by alexsherstinsky** [comment]
Closing -- the improvement that is the subject of this PR will be made when the multi-batch case is fully implemented.

**Pull request closed without merging**